From: David Bremner <david@tethera.net>
To: notmuch@notmuchmail.org
Subject: [PATCH] lib/message-file: close stream in destructor
Date: Thu,  9 May 2019 21:23:24 -0300
Message-Id: <20190510002324.31140-1-david@tethera.net>
X-Mailer: git-send-email 2.20.1

Without this,

$ make time-test OPTIONS=--small

leads to fatal errors from too many open files.

Thanks to st-gourichon-fid for bringing this problem to my attention in IRC.
---
 lib/message-file.c | 3 +++
 1 file changed, 3 insertions(+)

The regression test could be improved, but I wanted to get a fix out
and hopefully into master, as this is a problem that seems
particularly likely to hit new users during their initial notmuch-new.

diff --git a/lib/message-file.c b/lib/message-file.c
index 50855067..24c5fda4 100644
--- a/lib/message-file.c
+++ b/lib/message-file.c
@@ -46,6 +46,9 @@ _notmuch_message_file_destructor (notmuch_message_file_t *message)
     if (message->message)
 	g_object_unref (message->message);
 
+    if (message->stream)
+	g_object_unref (message->stream);
+
     return 0;
 }
 
-- 
2.20.1

_______________________________________________
notmuch mailing list
notmuch@notmuchmail.org
https://notmuchmail.org/mailman/listinfo/notmuch
